### PR TITLE
Use app specific password from Apple ID connection

### PR DESCRIPTION
**Pull request by adborbas**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _MINOR_ [version update](https://semver.org/)

### Context

Update appleauth package to support app specific password in Apple ID connection

Resolves: [STEP-613](https://bitrise.atlassian.net/browse/STEP-613)

### Changes

- Moved to `go mod` from `dep`
- Update `appleauth` package
- Added new test to bitrise.yml

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -3,165 +3,176 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
...
-    - path::./:
-        inputs:
-        - connection: apple_id
-        - altool_options: "--output-format xml --verbose"
-        - app_password: $APP_SPECIFIC_PASSWORD # App-specific password can not be set for the connection
+      - path::./:
+          inputs:
+            - connection: apple_id
+            - altool_options: "--output-format xml --verbose"
+            - app_password: $APP_SPECIFIC_PASSWORD # App-specific password can not be set for the connection
```
Comment (lpusok) [inline bitrise.yml:74]: The comment is no longer true.
Despite this test, the value of APP_SPECIFIC_PASSWORD will not be used as the connection contains it, so I rather remove this test than for it to not really test out the input.

Comment (lpusok) [reply bitrise.yml:74]: There is a way to test this right. Can use the BITRISE_BUILD_URL with a file scheme to add a custom json file. BITRISE_BUILD_API_TOKEN can contain anything.
In a script step, download the connection json, use jq to remove the application specific password and write to a file.

Comment (adborbas) [reply bitrise.yml:74]: I agree with the comment remove part, but I would keep the test to cover the case of having the `app_password` still set but the actual password being used from the connection. Basically to cover regression.

Comment (lpusok) [reply bitrise.yml:74]: In that case let's use a known invalid app specific password. Do not totally agree with the regression part as the positive case will not be tested.

Comment (godrei) [reply bitrise.yml:74]: I agree with Laci, and I would avoid adding more e2e test cases if possible (daily deploy limit issues).
I think the unit tests are covering the app-specific password selection.

Comment (adborbas) [reply bitrise.yml:74]: Ok, i'll remove this test then 👍

**Review by lpusok — APPROVED** [review]

**Pull request merged**